// dates-picker/dist/App.js
import React, { useState } from "react";
import "./App.css";
import "./styles/button.css";
import "./styles/date-range-picker-component.css";
import { InitialParametersProvider } from "./context/InitialParametersContext";
import { Button } from "./components/Button";
import { CalendarComponent } from "./components/CalendarComponent";
export function App(props) {
  const {
    callback,
    boardsNum: propsBoardsNum,
    startDate,
    endDate,
    defaultColor
  } = props;
  let boardsNumInitialState = propsBoardsNum;

  if (!propsBoardsNum) {
    boardsNumInitialState = 2;
  }

  const [boardsNum] = useState(boardsNumInitialState);
  const [showCalendar, setShowCalendar] = useState(false);
  const [buttonDatesText, setButtonDatesText] = useState(null);
  return /*#__PURE__*/React.createElement("div", {
    className: "App"
  }, /*#__PURE__*/React.createElement(InitialParametersProvider, {
    props: props
  }, /*#__PURE__*/React.createElement(Button, {
    showCalendar: showCalendar,
    setShowCalendar: setShowCalendar,
    buttonDatesText: buttonDatesText
  }), /*#__PURE__*/React.createElement(CalendarComponent, {
    callback: callback,
    boardsNum: boardsNum,
    showCalendar: showCalendar,
    setShowCalendar: setShowCalendar,
    buttonDatesText: buttonDatesText,
    setButtonDatesText: setButtonDatesText,
    startDate: startDate,
    endDate: endDate,
    defaultColor: defaultColor
  })));
}
