// dates-picker/dist/dist/actions.dev.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.setInitialBoard = setInitialBoard;
exports.setStoredDates = exports.setChosenDatesList = exports.setShowDaysAmountTab = exports.setChosenDates = exports.setFirstDayOfWeekIndex = exports.setEndDate = exports.setStartDate = exports.setLanguage = exports.setSelectedColor = exports.setMode = exports.setViewedYear = exports.setViewedMonth = exports.setShowColorPicker = void 0;

var _actionsUtils = require("./utils/actionsUtils");

var setShowColorPicker = function setShowColorPicker(boardsNum, id, showColorPicker) {
  return {
    type: "SET_SHOW_COLOR_PICKER",
    showColorPicker: showColorPicker,
    id: id,
    boardsNum: boardsNum
  };
};

exports.setShowColorPicker = setShowColorPicker;

var setViewedMonth = function setViewedMonth(boardsNum, id, viewedMonth) {
  return {
    type: "SET_VIEWED_MONTH",
    viewedMonth: viewedMonth,
    id: id,
    boardsNum: boardsNum
  };
};

exports.setViewedMonth = setViewedMonth;

var setViewedYear = function setViewedYear(boardsNum, id, viewedYear) {
  return {
    type: "SET_VIEWED_YEAR",
    viewedYear: viewedYear,
    id: id,
    boardsNum: boardsNum
  };
};

exports.setViewedYear = setViewedYear;

var setMode = function setMode(boardsNum, id, mode) {
  return {
    type: "SET_MODE",
    mode: mode,
    id: id,
    boardsNum: boardsNum
  };
};

exports.setMode = setMode;

var setSelectedColor = function setSelectedColor(selectedColor) {
  return {
    type: "SET_SELECTED_COLOR",
    selectedColor: selectedColor
  };
};

exports.setSelectedColor = setSelectedColor;

var setLanguage = function setLanguage(language) {
  return {
    type: "SET_LANGUAGE",
    language: language
  };
};

exports.setLanguage = setLanguage;

var setStartDate = function setStartDate(startDate) {
  return {
    type: "SET_START_DATE",
    startDate: startDate
  };
};

exports.setStartDate = setStartDate;

var setEndDate = function setEndDate(endDate) {
  return {
    type: "SET_END_DATE",
    endDate: endDate
  };
};

exports.setEndDate = setEndDate;

var setFirstDayOfWeekIndex = function setFirstDayOfWeekIndex(firstDayOfWeekIndex) {
  return {
    type: "SET_FIRST_DAY_OF_WEEK_INDEX",
    firstDayOfWeekIndex: firstDayOfWeekIndex
  };
};

exports.setFirstDayOfWeekIndex = setFirstDayOfWeekIndex;

function setInitialBoard(boardsNum, language, startDate, endDate) {
  return function (dispatch) {
    var _getInitialObject = (0, _actionsUtils.getInitialObject)(boardsNum, language, startDate, endDate),
        monthsObj = _getInitialObject.monthsObj,
        yearsObj = _getInitialObject.yearsObj,
        modeObj = _getInitialObject.modeObj;

    for (var id = 0; id < boardsNum; id++) {
      // dispatch(setBoardsNum(boardsNum));
      dispatch(setViewedMonth(boardsNum, id, monthsObj));
      dispatch(setViewedYear(boardsNum, id, yearsObj));
      dispatch(setMode(boardsNum, id, modeObj));
    }
  };
}

var setChosenDates = function setChosenDates(chosenDates) {
  return {
    type: "SET_CHOSEN_DATES",
    chosenDates: chosenDates
  };
};

exports.setChosenDates = setChosenDates;

var setShowDaysAmountTab = function setShowDaysAmountTab(showDaysAmountTab) {
  return {
    type: "SET_SHOW_DAYS_AMOUNT_TAB",
    showDaysAmountTab: showDaysAmountTab
  };
};

exports.setShowDaysAmountTab = setShowDaysAmountTab;

var setChosenDatesList = function setChosenDatesList(chosenDatesList) {
  return {
    type: "SET_CHOSEN_DATES_LIST",
    chosenDatesList: chosenDatesList
  };
};

exports.setChosenDatesList = setChosenDatesList;

var setStoredDates = function setStoredDates(storedDates) {
  return {
    type: "SET_STORED_DATES",
    storedDates: storedDates
  };
};

exports.setStoredDates = setStoredDates;
