// dates-picker/dist/components/CalendarModesComponents/CalendarContent.js
import React from "react";
import "../../styles/calendar-component.css";
import { WeekDaysNames } from "./WeekDaysNames";
import { DaysGrid } from "../DayElementsComponents/DaysGrid";
import { MonthSelector } from "./MonthSelector";
import { YearSelector } from "./YearSelector";
export const CalendarContent = props => {
  const {
    selectedColor,
    selectedDays,
    hoveredDay,
    setSelectedDays,
    setHoveredDay,
    calendarModesState,
    calendarModesStateDispatch,
    datesHeaderStateDispatch,
    boardsNum,
    datesHeaderState,
    nearViewedMonths,
    id
  } = props;
  const mode = calendarModesState.mode[id];
  return /*#__PURE__*/React.createElement("div", {
    className: "month-grid"
  }, /*#__PURE__*/React.createElement(WeekDaysNames, null), /*#__PURE__*/React.createElement(DaysGrid, {
    selectedColor: selectedColor,
    selectedDays: selectedDays,
    hoveredDay: hoveredDay,
    setSelectedDays: setSelectedDays,
    setHoveredDay: setHoveredDay,
    datesHeaderState: datesHeaderState,
    datesHeaderStateDispatch: datesHeaderStateDispatch,
    nearViewedMonths: nearViewedMonths,
    boardsNum: boardsNum,
    id: id
  }), mode === "Months" ? /*#__PURE__*/React.createElement(MonthSelector, {
    selectedColor: selectedColor,
    selectedDays: selectedDays,
    calendarModesStateDispatch: calendarModesStateDispatch,
    datesHeaderState: datesHeaderState,
    datesHeaderStateDispatch: datesHeaderStateDispatch,
    nearViewedMonthsFunction: nearViewedMonths,
    boardsNum: boardsNum,
    id: id
  }) : mode === "Years" && /*#__PURE__*/React.createElement(YearSelector, {
    selectedColor: selectedColor,
    selectedDays: selectedDays,
    calendarModesStateDispatch: calendarModesStateDispatch,
    datesHeaderState: datesHeaderState,
    datesHeaderStateDispatch: datesHeaderStateDispatch,
    boardsNum: boardsNum,
    nearViewedMonthsFunction: nearViewedMonths,
    id: id
  }));
};
