// dates-picker/dist/components/DatesHeaderComponents/DatesHeader.js
import React from "react";
import { calendarConfig } from "../../configuration/config";
import "../../styles/DatesHeaderStyles/dates-header.css";
import { useLanguage } from "../../context/InitialParametersContext";
import { InfoElement } from "./InfoElement";
import { Arrow } from "./Arrow";
export const DatesHeader = props => {
  const {
    selectedColor,
    calendarModesStateDispatch,
    datesHeaderState,
    datesHeaderStateDispatch,
    nearViewedMonths,
    boardsNum,
    id
  } = props;
  const viewedMonth = datesHeaderState.viewedMonth[id];
  const viewedYear = datesHeaderState.viewedYear[id];
  const language = useLanguage();
  const monthsElementValue = calendarConfig.months[language][viewedMonth];
  return /*#__PURE__*/React.createElement("div", {
    className: "header",
    lang: language
  }, /*#__PURE__*/React.createElement("div", {
    className: "info-elements"
  }, /*#__PURE__*/React.createElement(InfoElement, {
    element: "month",
    value: monthsElementValue,
    changeMode: "Months",
    selectedColor: selectedColor,
    boardsNum: boardsNum,
    calendarModesStateDispatch: calendarModesStateDispatch,
    id: id
  }), /*#__PURE__*/React.createElement(InfoElement, {
    element: "year",
    value: viewedYear,
    changeMode: "Years",
    selectedColor: selectedColor,
    boardsNum: boardsNum,
    calendarModesStateDispatch: calendarModesStateDispatch,
    id: id
  })), /*#__PURE__*/React.createElement("div", {
    className: "header-icons"
  }, /*#__PURE__*/React.createElement(Arrow, {
    nearViewedMonthsfunction: nearViewedMonths,
    datesHeaderState: datesHeaderState,
    datesHeaderStateDispatch: datesHeaderStateDispatch,
    selectedColor: selectedColor,
    boardsNum: boardsNum,
    id: id,
    arrowSide: "leftArrow"
  }), /*#__PURE__*/React.createElement(Arrow, {
    nearViewedMonthsfunction: nearViewedMonths,
    datesHeaderState: datesHeaderState,
    datesHeaderStateDispatch: datesHeaderStateDispatch,
    selectedColor: selectedColor,
    boardsNum: boardsNum,
    id: id,
    arrowSide: "rightArrow"
  })));
};
